app/gui/view/welcome-screen/src/lib.rs

```diff
@@ -51,6 +53,55 @@ mod css_id {
 
 
 
+// ========================
+// === ClickableElement ===
+// ========================
+
+
+// === ClickClosure ===
+
+/// Type alias for "on-click" event handlers on buttons.
+type ClickClosure = Closure<dyn FnMut(MouseEvent)>;
+
+
+// === ClickableElement ===
+
+/// Clickable HTML element. It has a single `click` event source that fires an event on each `click`
+/// JS event.
+#[derive(Debug, Clone, CloneRef)]
+struct ClickableElement {
+    pub element: Element,
+    pub click:   frp::Source,
+    pub network: frp::Network,
+}
+
+impl Deref for ClickableElement {
+    type Target = Element;
+    fn deref(&self) -> &Self::Target {
+        &self.element
+    }
+}
+
+impl ClickableElement {
+    pub fn new(element: Element, logger: &Logger) -> Self {
+        frp::new_network! { network
+            click <- source_();
+        }
+        let click_clone = click.clone_ref();
+        let closure = Box::new(move |_event: MouseEvent| {
+            click_clone.emit(());
+        });
+        let closure: ClickClosure = Closure::wrap(closure);
```

Comment:
use our macros instead: `let closure: ClickClosure = Closure::wrap(f_!(click.emit()))`

Reply:
And off my code is broken - there is another method, not `wrap` which automatically uses Box